engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/StreamProcessor.java

```diff
@@ -278,6 +278,8 @@ private ZeebeDbState recoverState() {
     final TransactionContext transactionContext = zeebeDb.createContext();
     final ZeebeDbState zeebeState = new ZeebeDbState(partitionId, zeebeDb, transactionContext);
 
+    zeebeState.registerLifecycleListeners(lifecycleAwareListeners::add);
```

Comment:
(optional) We could register the listener for the state like the other listeners in `EngineProcessors`. For example, if `ZeebeState` implements the listener interface. 

Currently, it should be okay to register the listener before. I don't see conflicts with the current migration steps. But it is a potential risk. The migration should happen before.